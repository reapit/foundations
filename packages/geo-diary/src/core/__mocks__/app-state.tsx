export const useAppState = jest.fn(() => {
  return {
    appState: {
      currentLat: null,
      currentLng: null,
      hasGeoLocation: false,
      time: 'TODAY',
      travelMode: 'DRIVING',
      destinationLat: null,
      destinationLng: null,
      appointmentId: null,
      appointment: null,
      tab: 'LIST',
      routeInformation: null,
      destinationAddress: null,
      locationAddress: null,
      mapRefs: null,
      locationQueryAddress: null,
      locationQueryResults: [],
      contactDrawerOpen: false,
      contactDrawerType: 'ATTENDEE',
      contactId: null,
      vendors: [],
      landlords: [],
      hasAmlApp: true,
      apptQuery: null,
    },
    setAppState: jest.fn(),
  }
})
