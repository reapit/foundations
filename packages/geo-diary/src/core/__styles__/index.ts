import { css } from 'linaria'

// layout
export const layoutBase = '1rem'
export const layoutQuarter = 'calc(1rem / 4)'
export const layoutThird = 'calc(1rem / 3)'
export const layoutHalf = 'calc(1rem / 2)'
export const layoutTwoThird = 'calc(1rem * 0.67)'
export const layoutThreeQuarter = 'calc(1rem * 0.75)'
export const layoutOneQuarter = 'calc(1rem * 1.25)'
export const layoutOneHalf = 'calc(1rem * 1.5)'
export const layoutDouble = 'calc(1rem * 2)'
export const layoutTwoHalf = 'calc(1rem * 2.5)'
export const layoutTriple = 'calc(1rem * 3)'
export const layoutQuadruple = 'calc(1rem * 4)'
export const layoutSextuple = 'calc(1rem * 6)'
// color
export const white = '#fff'
export const black = '#12263f'
export const greyDark = '#6e84a3'
export const grey = '#95aac9'
export const greyLighter = '#e3ebf6'
export const greyLightest = '#f9fbfd'
export const green = '#5fe781'
export const greenLighter = '#acf2bd'
export const greenLightest = '#e6ffed'
export const red = '#e96171'
export const redLighter = '#fdb8c0'
export const redLightest = '#ffeef0'
export const reapitDarkblue = '#262f69'
export const reapitMidblue = '#0061a8'
export const reapitLightblue = '#23a4de'
export const reapitLightestblue = '#7bc9eb'
export const reapitOrange = '#ec631b'
export const reapitLime = '#cddb00'
export const reapitTeal = '#006580'
export const reapitPlumb = '#7a2c81'
export const reapitPurple = '#a4185c'
export const reapitGold = '#ffb71b'
export const reapitGreen = '#a0c862'
export const reapitRed = '#d3033d'

export const globals = css`
  :global() {
    @font-face {
      font-family: 'Museo Sans W03_300';
      src: url('../../assets/fonts/1427936/f2cf6cf9-9ec6-4945-a525-f5873d143c2a.eot?#iefix');
      src: url('../../assets/fonts/1427936/f2cf6cf9-9ec6-4945-a525-f5873d143c2a.eot?#iefix') format('eot'),
        url('../../assets/fonts/1427936/9803fddf-c005-431a-92d5-0f18688f945d.woff2') format('woff2'),
        url('../../assets/fonts/1427936/ea9b8ac3-ff16-4387-a473-32a6a617329f.woff') format('woff'),
        url('../../assets/fonts/1427936/353acc2c-88f2-4de3-83eb-6cc2c9b05af1.ttf') format('truetype');
    }
    @font-face {
      font-family: 'Museo Sans W03_700';
      src: url('../../assets/fonts/1427944/1c070cdb-18d8-440e-be9d-2448fa3930c4.eot?#iefix');
      src: url('../../assets/fonts/1427944/1c070cdb-18d8-440e-be9d-2448fa3930c4.eot?#iefix') format('eot'),
        url('../../assets/fonts/1427944/66c06801-da3e-4587-a89c-674cfbe39c21.woff2') format('woff2'),
        url('../../assets/fonts/1427944/f3c7f613-9728-4ed6-a383-1c8519b215d2.woff') format('woff'),
        url('../../assets/fonts/1427944/c21bf502-6b58-4bf0-9ddd-169929c263e7.ttf') format('truetype');
    }
  }
  @import url('https://fonts.googleapis.com/css?family=Source+Code+Pro&display=swap');
  @import url('https://fonts.googleapis.com/css?family=Roboto&display=swap');

  #root {
    height: 100%;
  }
`
