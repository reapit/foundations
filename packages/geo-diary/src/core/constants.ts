export const DEFAULT_LAT_LNG = {
  lat: 51.509865,
  lng: -0.118092,
}

export const DEFAULT_ZOOM = 10
