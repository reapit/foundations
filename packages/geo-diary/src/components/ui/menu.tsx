import * as React from 'react'
import { useDispatch } from 'react-redux'
import { useLocation } from 'react-router'
import { Dispatch } from 'redux'
import { Menu as Sidebar, MenuConfig, ReapitLogo } from '@reapit/elements'
import { authLogout } from '@/actions/auth'
import { Location } from 'history'
import { FaSignOutAlt, FaCloud, FaMapMarkerAlt } from 'react-icons/fa'
import { ActionCreator } from '@/types/core'

export const generateMenuConfig = (logoutCallback: () => void, location: Location<any>): MenuConfig => {
  return {
    defaultActiveKey: 'GEO_DIARY',
    location,
    menu: [
      {
        key: 'LOGO',
        icon: <ReapitLogo className="nav-item-icon" />,
        type: 'LOGO',
      },
      {
        title: 'Geo Diary',
        key: 'GEO_DIARY',
        icon: <FaMapMarkerAlt className="nav-item-icon" />,
        url: '/',
        type: 'PRIMARY',
      },
      {
        title: 'Apps',
        key: 'APPS',
        icon: <FaCloud className="nav-item-icon" />,
        callback: () =>
          (window.location.href =
            window.location.href.includes('dev') || window.location.href.includes('localhost')
              ? 'https://dev.marketplace.reapit.cloud/client/installed'
              : 'https://marketplace.reapit.cloud/client/installed'),
        type: 'PRIMARY',
      },
      {
        title: 'Logout',
        key: 'LOGOUT',
        callback: logoutCallback,
        icon: <FaSignOutAlt className="nav-item-icon" />,
        type: 'SECONDARY',
      },
    ],
  }
}

export interface MenuMappedActions {
  logout: () => void
}

export interface MenuMappedState {}

export type MenuProps = {}

export const logout = ({ dispatch, authLogout }: { dispatch: Dispatch; authLogout: ActionCreator<void> }) => () =>
  dispatch(authLogout())

export const Menu: React.FunctionComponent<MenuProps> = () => {
  const location = useLocation()
  const dispatch = useDispatch()
  const menuConfigs = generateMenuConfig(logout({ dispatch, authLogout }), location)
  return <Sidebar {...menuConfigs} location={location} />
}

export default Menu
