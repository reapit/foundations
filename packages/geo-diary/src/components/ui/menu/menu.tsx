import * as React from 'react'
import { useLocation } from 'react-router'
import { Menu as Sidebar, MenuConfig, ReapitLogo } from '@reapit/elements'
import { Location } from 'history'
import { FaSignOutAlt, FaCloud } from 'react-icons/fa'
import { useReapitConnect } from '@reapit/connect-session'
import { reapitConnectBrowserSession } from '@/core/connect-session'

export const generateMenuConfig = (logoutCallback: () => void, location: Location<any>): MenuConfig => {
  return {
    defaultActiveKey: 'APPS',
    location,
    menu: [
      {
        key: 'LOGO',
        icon: <ReapitLogo className="nav-item-icon" />,
        type: 'LOGO',
      },
      {
        title: 'Apps',
        key: 'APPS',
        icon: <FaCloud className="nav-item-icon" />,
        callback: callbackAppClick,
        type: 'PRIMARY',
      },
      {
        title: 'Logout',
        key: 'LOGOUT',
        callback: logoutCallback,
        icon: <FaSignOutAlt className="nav-item-icon" />,
        type: 'SECONDARY',
      },
    ],
  }
}

export const callbackAppClick = () =>
  (window.location.href =
    window.location.href.includes('dev') || window.location.href.includes('localhost')
      ? 'https://marketplace.dev.paas.reapit.cloud/installed'
      : 'https://marketplace.reapit.cloud/installed')

export type MenuProps = {}

export const Menu: React.FC<MenuProps> = () => {
  const location = useLocation()
  const { connectLogoutRedirect, connectIsDesktop } = useReapitConnect(reapitConnectBrowserSession)
  const menuConfigs = generateMenuConfig(() => connectLogoutRedirect(), location)
  const desktopOptimisedMenu = connectIsDesktop
    ? {
        ...menuConfigs,
        menu: menuConfigs.menu.filter(config => config.key !== 'APPS' && config.key !== 'LOGOUT'),
      }
    : menuConfigs
  return <Sidebar {...desktopOptimisedMenu} location={location} />
}

export default Menu
