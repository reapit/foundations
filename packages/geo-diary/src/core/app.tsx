import * as React from 'react'
import Router from './router'
import './__styles__'

const App = () => {
  return <Router />
}

export default App
