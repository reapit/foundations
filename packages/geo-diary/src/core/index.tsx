import React from 'react'
import * as Sentry from '@sentry/browser'
import { render } from 'react-dom'
import ReactGA from 'react-ga'
import { Config } from '@/types/global'
import App from './app'
import { register } from './service-worker'
import '../styles/index.scss'
import { getSessionCookie } from '@reapit/cognito-auth'
import { COOKIE_SESSION_KEY_GEO_DIARY } from '../constants/api'
import store from './store'
import { authSetRefreshSession } from '../actions/auth'

// Init global config
window.reapit = {
  config: {
    appEnv: 'production',
    sentryDns: '',
    platformApiUrl: '',
    cognitoClientId: '',
    googleAnalyticsKey: '',
    googleMapApiKey: '',
    cognitoOAuthUrl: '',
    cognitoUserPoolId: '',
  },
}

export const renderApp = (Component: React.ComponentType) => {
  const rootElement = document.querySelector('#root') as Element
  const refreshSessionFromCookie = getSessionCookie(COOKIE_SESSION_KEY_GEO_DIARY, window.reapit.config.appEnv)
  if (refreshSessionFromCookie) {
    store.dispatch(authSetRefreshSession(refreshSessionFromCookie))
  }
  if (rootElement) {
    render(<Component />, rootElement)
  }
}

const run = async () => {
  await fetch('config.json')
    .then(response => response.json())
    .then((config: Config) => {
      window.reapit.config = config
      const isLocal = config.appEnv === 'local'
      if (!isLocal && config.sentryDns) {
        Sentry.init({
          release: process.env.APP_VERSION,
          dsn: config.sentryDns,
          environment: config.appEnv,
        })
      }
      if (!isLocal && config.googleAnalyticsKey) {
        ReactGA.initialize(config.googleAnalyticsKey)
        ReactGA.pageview(window.location.pathname + window.location.search)
      }
      renderApp(App)
    })
    .catch(error => {
      console.error('Cannot fetch config', error)
    })
}

if (module['hot']) {
  module['hot'].accept('./app', () => {
    const NextApp = require('./app').default
    renderApp(NextApp)
  })
}

if (process.env.NODE_ENV === 'production') {
  register()
}

run()
