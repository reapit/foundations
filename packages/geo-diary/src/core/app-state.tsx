/* istanbul ignore file */
/* Had to add because of skipped test, react hooks testing not yet supporting React 18 
https://github.com/testing-library/react-hooks-testing-library/issues/654 can remove when tests un-skipped
Looks like we will have to migrate to the main testing lib when this PR is merged
https://github.com/testing-library/react-testing-library/pull/991*/
import React, { createContext, Dispatch, SetStateAction, useContext, useEffect, useState } from 'react'
import { VendorLandlordModel } from '../components/pages/appointment/appointment'
import { ContactDrawerType } from '../components/ui/contact-drawer'
import { Distance, Duration, GeocoderResult, MapRefs } from '../components/ui/map/types'
import { ExtendedAppointmentModel } from '../types/global'
import { getGeoCoords } from '../utils/map-utils'

export type AppTimeRange = 'TODAY' | 'TOMORROW' | 'WEEK'
export type AppTravelMode = 'DRIVING' | 'WALKING'
export type AppTab = 'MAP' | 'LIST'
export interface AppRouteInformation {
  duration: Duration
  distance: Distance
}

export interface AppState {
  currentLat: number | null
  currentLng: number | null
  hasGeoLocation: boolean
  time: AppTimeRange
  travelMode: AppTravelMode
  destinationLat: number | null
  destinationLng: number | null
  appointmentId: string | null
  appointment: ExtendedAppointmentModel | null
  tab: AppTab
  routeInformation: AppRouteInformation | null
  destinationAddress: string | null
  locationAddress: string | null
  mapRefs: MapRefs | null
  locationQueryAddress: string | null
  locationQueryResults: GeocoderResult[]
  contactDrawerOpen: boolean
  contactDrawerType: ContactDrawerType
  contactId: string | null
  vendors: VendorLandlordModel[]
  landlords: VendorLandlordModel[]
  hasAmlApp: boolean
}

export interface AppStateContextProps {
  appState: AppState
  setAppState: Dispatch<SetStateAction<AppState>>
}

export const defaultAppState: AppState = {
  currentLat: null,
  currentLng: null,
  hasGeoLocation: false,
  time: 'TODAY',
  travelMode: 'DRIVING',
  destinationLat: null,
  destinationLng: null,
  appointmentId: null,
  appointment: null,
  tab: 'LIST',
  routeInformation: null,
  destinationAddress: null,
  locationAddress: null,
  mapRefs: null,
  locationQueryAddress: null,
  locationQueryResults: [],
  contactDrawerOpen: false,
  contactDrawerType: ContactDrawerType.ATTENDEE,
  contactId: null,
  vendors: [],
  landlords: [],
  hasAmlApp: false,
}

export const AppStateContext = createContext<AppStateContextProps>({} as AppStateContextProps)

const { Provider } = AppStateContext

export const AppStateProvider: React.FC = ({ children }) => {
  const [appState, setAppState] = useState<AppState>(defaultAppState)

  useEffect(() => {
    const getAppState = async () => {
      const geoCoords = await getGeoCoords()
      setAppState((currentState) => ({
        ...currentState,
        ...geoCoords,
      }))
    }
    getAppState()
  }, [])

  return (
    <Provider
      value={{
        appState,
        setAppState,
      }}
    >
      {children}
    </Provider>
  )
}

export const useAppState = () => {
  const { appState, setAppState } = useContext(AppStateContext)
  return {
    appState,
    setAppState,
  }
}
