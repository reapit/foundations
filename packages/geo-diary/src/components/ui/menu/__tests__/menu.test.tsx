import * as React from 'react'
import { shallow } from 'enzyme'
import { Menu, generateMenuConfig } from '../menu'
import { AppState } from '../../../../core/app-state'

jest.mock('react-router', () => ({
  ...(jest.requireActual('react-router') as Object),
  useLocation: jest.fn(() => ({
    location: 'location',
  })),
}))

jest.mock('../../../../core/app-state')

describe('Menu', () => {
  it('should match a snapshot', () => {
    const wrapper = shallow(<Menu />)
    expect(wrapper).toMatchSnapshot()
  })

  describe('generateMenuConfig', () => {
    it('should return config', () => {
      const location = {
        hash: 'mockHash',
        key: 'mockKey',
        pathname: 'mockPathname',
        search: '',
        state: {},
      }
      const logoutCallback = jest.fn()
      const setAppState = jest.fn()
      const setPwaNavState = jest.fn()
      const result = generateMenuConfig(logoutCallback, location, setAppState, setPwaNavState, {} as AppState, true)
      expect(result).toBeDefined()
    })
  })
})
