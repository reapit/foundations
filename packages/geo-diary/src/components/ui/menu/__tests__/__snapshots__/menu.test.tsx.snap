// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match a snapshot 1`] = `
<Memo(MenuComponent)
  defaultActiveKey="APPS"
  location={
    Object {
      "location": "location",
    }
  }
  menu={
    Array [
      Object {
        "icon": <Unknown
          className="nav-item-icon"
        />,
        "key": "LOGO",
        "type": "LOGO",
      },
      Object {
        "callback": [Function],
        "icon": <FaCloud
          className="nav-item-icon"
        />,
        "key": "APPS",
        "title": "Apps",
        "type": "PRIMARY",
      },
      Object {
        "callback": [Function],
        "icon": <FaSignOutAlt
          className="nav-item-icon"
        />,
        "key": "LOGOUT",
        "title": "Logout",
        "type": "SECONDARY",
      },
    ]
  }
/>
`;
